vterebey/budgetcalculator/budget_calculator_class.py

```diff
@@ -0,0 +1,302 @@
+            writer_.writerow([self.category, self.date, self.amount])
+        Budget._add_expense(self)
+
+    def expenses_stat(self):
+        """This function dispays expenses in the form of a chart"""
+        amounts = list(Budget.categories.values())
+        # pylint: disable=W0612
+        fig1, ax1 = plt.subplots()
+        ax1.pie(
+            amounts,
+            labels=list(Budget.categories.keys()),
+            autopct="%1.1f%%",
+            shadow=False,
+            startangle=90,
+        )
+        ax1.axis("equal")
+        ax1.set_title("Statistics")
+        plt.show()
+
+    def group_by_cat(self, category):
+        """This function groups expenses by category"""
+        with open("values.csv", "r", encoding="UTF-8", newline="") as csvfile:
+            reader = csv.DictReader(csvfile, delimiter=",")
+            print(*fieldnames, sep=",")
+            for row in reader:
+                if row["Category"] == category:
+                    print(f"{row['Category']},{row['Date']},{row['Amount']}")
+        print()
+
+    def group_by_month(self, month):
+        """This function groups expenses by month"""
+        with open("values.csv", "r", encoding="UTF-8", newline="") as csvfile:
+            reader = csv.DictReader(csvfile, delimiter=",")
+            print(*fieldnames, sep=",")
+            for row in reader:
+                check_month = datetime.strptime(row["Date"], "%Y-%m-%d")
+                check_month = check_month.strftime("%Y-%m")
+                if check_month == month:
+                    print(f"{row['Category']},{row['Date']},{row['Amount']}")
+        print()
+
+    def group_by_cat_month(self, category, month):
+        """This function groups expenses by category and month"""
+        with open("values.csv", "r", encoding="UTF-8", newline="") as csvfile:
+            reader = csv.DictReader(csvfile, delimiter=",")
+            print(*fieldnames, sep=",")
+            for row in reader:
+                check_month = datetime.strptime(row["Date"], "%Y-%m-%d")
+                check_month = check_month.strftime("%Y-%m")
+                if check_month == month and row["Category"] == category:
+                    print(f"{row['Category']},{row['Date']},{row['Amount']}")
+        print()
+
+
+def welcome_menu():
+    """This function displays a menu"""
+    print(
+        """What would like to do:
+1. Make an entry
+2. See statistics
+3. Group by category
+4. Group by month
+5. Group by category and month
+6. Exit"""
+    )
+
+
+def options_menu():
+    """
+    This function gives the choice to choose a category
+    by itself or from the default categories
+    """
+    input_valid = False
+    while not input_valid:
+        try:
+            input_choice = int(
+                input("""Enter 1 to select a category or 2 to enter\n""")
+            )
+            if input_choice == 1:
+                input_valid = True
+                print(
+                    """
+1. Grocery store
```

Comment:
Let's make this print generic, based on available categories

Reply:
Updated